### ISSUE TITLE
F11 Order List & Panning/Channel Volume: tapping F11 2x to show Channel Volume doesn't work (OSX)

### ISSUE THREAD

**Issue by kserp** [schismtracker/schismtracker#358]
MacBook Pro (13-inch, Late 2011)
OSX 10.12.6
QWERTY

MacBook Pro (15-inch, 2017)
OSX 12.6
QWERTY

Schism Tracker 20221201

Title is explanatory I think, I believe tapping F11 twice should toggle between Panning & Channel Volume, which doesn't seem to work on OSX.

**Comment by abandpass** [comment]
I can reproduce this on an m1 mac mini (2020) on mac os 13.2.1
Tested an older version : 20220905 on mac os and pressing F11 twice works as expected.

**Comment by pedrohex** [comment]
Please remember that some Function keys in mac are associated with desktop shortcuts and override SchismTracker.
If I'm not wrong, F11 is associated witht the "Show Desktop" function. Try disable those configurations and try again.

**Comment by andikulla** [comment]
The issue is due to the fact that the function `set_page()` is always called twice on mac: the first time in `page.c`, the second time in the (very) long switch-case statement at the case`SCHISM_EVENT_NATIVE_SCRIPT` in `main.c`. The last function call [in the main.c explicitly calls the panning-page](https://github.com/schismtracker/schismtracker/blob/bc3c8c649a73dc400710def3d9b8e81d7200355d/schism/main.c#L922). Commenting out the line resolves the issue but I think the root cause probably lies in the file [macosx-sdlmain.m](https://github.com/schismtracker/schismtracker/blob/bc3c8c649a73dc400710def3d9b8e81d7200355d/sys/macosx/macosx-sdlmain.m)

**Comment by andikulla** [comment]
Dug a little bit deeper. As expected the issue lies in [macosx-sdlmain.m](https://github.com/schismtracker/schismtracker/blob/bc3c8c649a73dc400710def3d9b8e81d7200355d/sys/macosx/macosx-sdlmain.m): the file adds a macos menu item together with the macos shortcut F11. When F11 is pressed, schismtracker interprets it first in `page.c` and then the `SDL_WaitEvent` while loop in the `main.c` catches the MacOS F11 shortcut (which explicitly sets it to the panning page). Removing the shortcut for F11 works as a workaround/quick fix:
```diff
diff --git a/sys/macosx/macosx-sdlmain.m b/sys/macosx/macosx-sdlmain.m
index b1013072..a85cd6e7 100644
--- a/sys/macosx/macosx-sdlmain.m
+++ b/sys/macosx/macosx-sdlmain.m
@@ -185,7 +185,7 @@ static void setApplicationMenu(void)
         [menuItem setRepresentedObject: @"pattern"];
         menuItem = (NSMenuItem*)[appleMenu addItemWithTitle:@"Orders/Panning"
                                 action:@selector(_menu_callback:)
-                                keyEquivalent:KEQ_FN(11)];
+                                keyEquivalent: @""];
         [menuItem setKeyEquivalentModifierMask:0];
         [menuItem setRepresentedObject: @"orders"];
         menuItem = (NSMenuItem*)[appleMenu addItemWithTitle:@"Variables"
```

**Comment by abandpass** [comment]
I can confirm andikulla's workaround fixes the issue for me under mac os 13.6.  Thanks a lot!

**Comment by esaruoho** [comment]
@andikulla could you make a PR out of that?

**Comment by kserp** [comment]
Would that also fix the F2 Dialog overlay issue in #363?

**Comment by andikulla** [comment]
Sure, will do when I find the time in the next days. I will probably simply remove all keyEquivalent entries since they cause issues in general and aren't really necessary (except for the key hints in the menu bar)

**Comment by andikulla** [comment]
> Would that also fix the F2 Dialog overlay issue in #363?

Just checked, yes it does. [PR is open](https://github.com/schismtracker/schismtracker/pull/387) 😊
